ntoskrnl/io/pnpmgr/pnpres.c

```diff
@@ -542,12 +555,17 @@ IopCheckResourceDescriptor(
    IN BOOLEAN Silent,
    OUT OPTIONAL PCM_PARTIAL_RESOURCE_DESCRIPTOR ConflictingDescriptor)
 {
-   ULONG i, ii;
+   ULONG i, ii, DescriptorSize;
    BOOLEAN Result = FALSE;
+   PCM_FULL_RESOURCE_DESCRIPTOR FullDesc;
 
+   FullDesc = &ResourceList->List[0];
    for (i = 0; i < ResourceList->Count; i++)
    {
-      PCM_PARTIAL_RESOURCE_LIST ResList = &ResourceList->List[i].PartialResourceList;
+      PCM_PARTIAL_RESOURCE_LIST ResList = &FullDesc->PartialResourceList;
```

Comment:
Doing this correctly broke boot on x86 last I tried. https://jira.reactos.org/browse/CORE-10146

Reply:
It still does. Or rather, after hacking hal it boots, but the i8042prt driver doesn't load due to resource conflict. Any help figuring out why that is / fixing it is welcome.

Reply:
I noticed a while ago that our acpi.sys doesn't correctly translate the resource info from the ACPI tables either, and suspected that could be part of the problem. I have a patch somewhere that's a start on testing/fixing this, but it's unfinished, as other priorities came up. That was what I'd planned as the next step, though I don't know if it will even fix anything.

Reply:
@tkreuzer : about which hack hal?
Can add debugging information to find the cause of the conflict.

Reply:
@vgalnt Believe me, when I think some debug print information helps, I either add that, or manually step though the code in WinDbg. The info is all there, it just needs someone to correctly interprete it.